src/Proxy.hs

```diff
@@ -1698,65 +1758,65 @@ instance FromJSON DropSchedule where
       return Normal
     else fail "Unsupported drop type."
 
--- | Handle a GTU drop request.
---     When 'gtuDropData' is provided: handle drop.
---     Otherwise: return 404.
+-- |Return a handler which performs a GTU drop to the specified account.
+-- Attempt to make a GTU drop to the account with the specified address if @gtuDropData@ was provided
+-- in the environment settings. Responds with a 404-status code if @gtuDropData@ was absent.
 putGTUDropR :: Text -> Handler TypedContent
 putGTUDropR addrText = do
-    i <- internationalize
     Proxy{..} <- getYesod
     case gtuDropData of
       Nothing -> respond404Error $ EMErrorResponse NotFound
       Just gtuDropData' -> do
-        case addressFromText addrText of
-          Left _ -> respond400Error EMMalformedAddress RequestInvalid
-          Right addr -> runGRPC (doGetAccInfo addrText) $ \case
-              -- Account is not finalized
-              Nothing -> sendResponseStatus notFound404 $ object
-                          ["errorMessage" .= i18n i EMAccountNotFinal,
-                          "error" .= fromEnum RequestInvalid]
-              -- Account is finalized, so try the drop
-              Just _ -> do
-                connect rawRequestBody (sinkParserEither json') >>= \case
-                  Left _ -> tryDrop addr Normal gtuDropData' -- malformed or empty body, we assume normal drop.
-                  Right v -> case fromJSON v of
-                              AE.Success x -> tryDrop addr x gtuDropData'
-                              AE.Error e -> do
-                                $(logWarn) (Text.pack e)
-                                respond400Error EMConfigurationError RequestInvalid
-
+        addr <- doParseAccountAddress "putGTUDrop" addrText
+        -- Check that the account exists in a finalized block.
+        runGRPC (getAccountInfo (AccAddress addr) LastFinal) $ \_ -> do
```

Comment:
Won't this fail with NotFound error if the account does not exist instead of InvalidRequest which it did before?